### PR TITLE
Migrate tonal-interval to typescript

### PR DESCRIPTION
**Pull request by rakannimer**
## What does this PR do?

This is dependent on #73. Only [this commit](https://github.com/danigb/tonal/pull/79/commits/36377c5b930f568747f70953f210db249cf7a95b) is new.

Migrates tonal-interval to typescript. 

Centralizes jsdoc2md configuration to root directory.

Increases tonal-interval test coverage to 100%.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Where should a reviewer start?

This PR comes after #73 

## How has this been tested? How can we test it?

Tested locally, waiting for CI results. [EDIT : works on CI too 😄 ]

## Checklist:
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/interval/index.ts [hunk 1]
```diff
@@ -1,3 +1,38 @@
+type IntervalName = string;
+type IntervalQuality = string;
```
Comment (danigb) [inline packages/interval/index.ts]: I think this case can be narrowed:
```suggestion
type IntervalQuality = "dddd" | "ddd" | "dd" | "d" | "m" | "M" | "P" | "A" | "AA" | "AAA" | "AAAA";
```

File: packages/interval/index.ts [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+type IntervalAlteration = number;
+type IntervalStep = number;
+type IntervalDirection = 1 | -1;
+type IntervalSimplifiedNumber = number;
```
Comment (danigb) [inline packages/interval/index.ts]: This is a number from 1 to 8 (both included)... Can be modeled?

File: packages/interval/index.ts [hunk 3]
```diff
@@ -1,3 +1,38 @@
...
+type IntervalType = "P" | "M";
+type IntervalChroma = number;
+type IntervalClass = string;
```
Comment (danigb) [inline packages/interval/index.ts]: A number between 0 and 5 ...

File: packages/interval/index.ts [hunk 4]
```diff
@@ -1,3 +1,38 @@
+type IntervalName = string;
+type IntervalQuality = string;
+type IntervalAlteration = number;
+type IntervalStep = number;
+type IntervalDirection = 1 | -1;
+type IntervalSimplifiedNumber = number;
+type IntervalType = "P" | "M";
+type IntervalChroma = number;
```
Comment (danigb) [inline packages/interval/index.ts]: From docs: "The chroma is a number between 0 and 7 (...)" ... does have any impact on the type?

File: packages/interval/index.ts [hunk 5]
```diff
@@ -239,13 +292,16 @@ export const ic = ivl => {
  * Interval.build({ step: 1, alt: -1, oct: 0, dir: 1 }) // => "1d"
  * Interval.build({ num: 9, alt: -1 }) // => "9m"
  */
-export const build = ({ num, step, alt, oct = 1, dir } = {}) => {
+export const build = (
+  { num, step, alt, oct = 1, dir } = {} as Partial<IntervalProps>
```
Comment (danigb) [inline packages/interval/index.ts]: Nice!

**Comment by danigb** [comment]
This one looks really good too! 👏 🎉 Awesome job @rakannimer, I think is ready to merge... do I missing anything?

**Comment by danigb** [comment]
Another one! 🎉  Thanks! 🙏  

I've updated master and I've moved the jsdoc2md.json config file inside `config/` folder. 

To practice myTypeScript, I'm going to migrate tonal-distance module (and I'll add you as reviewer my PR, if possible). Of course, more PRs with module migrations are highly welcomed! 🏄🏻‍♀️

**Comment by rakannimer** [comment]
Awesome 🎉 ! 

Yeah for sure, happy to review feel free to add me to PRs 👍 

Cheers !

**Pull request merged**